precompiles/staking/testdata/StakingCallerWithFailure.sol

```diff
@@ -0,0 +1,580 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17;
+
+import "../StakingI.sol" as staking;
+
+/// @title StakingCaller
+/// @author Evmos Core Team
+/// @dev This contract is used to test external contract calls to the staking precompile.
+contract StakingCallerWithFailure {
```

Comment:
how come this contract was newly added, seems unrelated to the rest of the changes in the PR?

Reply:
good catch!